### DISCUSSION TITLE
Want to contribute socket.io adapter implemented with NATS.io

### DISCUSSION THREAD

**Discussion by distrue** [socketio/socket.io#4216] [Ideas] [answered]
Thanks for such a nice project `socket.io`! Recently I have built socket project with socket.io and [socket.io-redis-adapter](https://github.com/socketio/socket.io-redis-adapter). 

And I have reached to an limitation of building high-availability messaging system since redis single machine is bottleneck of the architecture. I considered redis-cluster to get rid of the problem, but it was unsuccessful to meet my requirement (1) easily auto-scalable and (2) well maintained messaging system client. In the midst of these concerns, I found [NATS.io](https://nats.io/) to alternate redis. 

Now I am building socket.io-adapter using NATS.io as messaging service. 
- repo: https://github.com/distrue/socket.io-nats-adapter
- npm: https://www.npmjs.com/package/socket.io-nats-adapter

The project probably be finished within this month, and I want to share the result with the community. Is there a process for sharing project to socket.io community?

**Comment by darrachequesne** [comment, marked as answer]
That's great, thanks! I think we could list it in the adapter section here: https://socket.io/docs/v4/adapter/

You can also take a look at https://github.com/MickL/socket.io-nats-adapter, for reference.

**Reply by distrue** [reply]
Thanks! I'll share again when adapter spec is on ready

**Reply by distrue** [reply]
@darrachequesne  It's been a while since I've shared news.
I finally finished the work and created an issue, it would be glad to check it out~!
https://github.com/socketio/socket.io/issues/4294
